### PR TITLE
General code fixes

### PR DESCRIPTION
**Pull request by michalbundyra**
- updated PHP CodeSniffer to version 2.7
- added phpcs.xml with defined CS rules
- fixed code to pass all CS checks
- short array syntax in docs
- composer scripts to run unit and cs checks
- updated travis configuration to use composer scripts

ping @gianarb

### CODE REVIEW

File: phpcs.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
```
Comment (gianarb) [inline phpcs.xml]: If I remember well we can point `phpcs` into the correct configuration file without copy it out.
Why did you move it here?
The idea is use the original one downloaded by composer in order to be up to date with the original one

**Review by gianarb — APPROVED** [review]

File: phpcs.xml (hunk 1 above)
Comment (webimpress) [inline phpcs.xml]: The same way we have in Zend/Apigility, for example:
https://github.com/zendframework/zend-inputfilter/blob/master/phpcs.xml
https://github.com/zfcampus/zf-rest/blob/master/phpcs.xml
This is because we define more rules here, not only PSR-2, but also forcing to use short array syntax, white spaces rules... etc.
Also in this file we have defined paths to check and excluded files (could be different for each project).

**Comment by alcaeus** [comment]
Superseded by #184 and #186.
